internal/torcx/types.go

```diff
@@ -81,6 +83,56 @@ type Image struct {
 	Remote    string `json:"remote"`
 }
 
+// ArchiveFormat is a torcx archive format, either 'tgz' or 'squashfs'
+type ArchiveFormat string
+
+const (
+	// ArchiveFormatUnknown is the zero value of ArchiveFormat. It indicates the image format is unknown
+	ArchiveFormatUnknown ArchiveFormat = ""
+	// ArchiveFormatTgz indicates a tar-gzipped image
+	ArchiveFormatTgz = "tgz"
+	// ArchiveFormatSquashfs indicates a squashfs image archive
+	ArchiveFormatSquashfs = "squashfs"
+)
+
+// UnmarshalJSON unmarshals an ArchiveFormat
+func (arf *ArchiveFormat) UnmarshalJSON(b []byte) error {
+	s := ""
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+	switch s {
+	case ArchiveFormatTgz:
+		*arf = ArchiveFormatTgz
+	case ArchiveFormatSquashfs:
+		*arf = ArchiveFormatSquashfs
+	default:
+		return fmt.Errorf("could not unmarshal into ArchiveFormat: must be one of %q, %q", ArchiveFormatTgz, ArchiveFormatSquashfs)
+	}
+	return nil
+}
+
+// UnmarshalJSON unmarshals an Archive, including defaulting the "format" field
+// to "tgz" if it was not set.
+func (ar *Archive) UnmarshalJSON(b []byte) error {
+	type archiveAlias Archive
+	var arAlias archiveAlias
+	if err := json.Unmarshal(b, &arAlias); err != nil {
+		return err
+	}
+	if arAlias.Format == ArchiveFormatUnknown {
+		// Default to tgz if it wasn't set
+		arAlias.Format = ArchiveFormatTgz
+	}
+	*ar = Archive(arAlias)
+	return nil
+}
+
+// FileSuffix returns the file extension this archive format must have.
+func (arf ArchiveFormat) FileSuffix() string {
+	return fmt.Sprintf(".torcx.%s", arf)
```

Comment:
Should we default the zero value to `tgz` in here? Or take care of it in some other ways?

Reply:
The intent is that the zero value is never used, so I'd rather either ignore it here or panic here.

Defaulting to `tgz` should happen a layer higher. Are you okay with it just being the caller's responsibility, or would you like me to panic / return an error / handle it here?

Reply:
I guess it's a matter of taste then, so I won't insist. I slightly prefer returning a string+error tuple, but I'm also fine with leaving it to the caller (as is).